src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -47,7 +45,19 @@ class SyncController(
   override def receive: Receive = idle
 
   def idle: Receive = handlePeerUpdates orElse {
-    case StartFastSync =>
+
+    case StartSync =>
+      (appStateStorage.isFastSyncDone(), doFastSync) match {
+        case (false, true) =>
+          self ! StartFastSync
```

Comment:
actually behavior seems to be correct - there was no fast sync state to resume from so when fast sync started it started downloading blocks from genesis
but we can add check if there is any saved state to resume (`fastSyncStateStorage.getSyncState().isEmpty`) and if there is no state then wipe db before starting fast sync
@mcsherrylabs what do you think?